### PR TITLE
Add heap reporting and limiting ability

### PR DESCRIPTION
**Pull request by gwicke**
https://phabricator.wikimedia.org/T89932

### CODE REVIEW

**Comment by gwicke** [comment]
Heap graph:
image

File: lib/heapwatch.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    // Report memory stats to statsd. Use 'timing' (which isn't really
+    // timing-specific at all) instead of 'gauge' to get percentiles &
+    // min/max.
+    var statsd = this.statsd;
+    statsd.timing('heap.rss', usage.rss);
+    statsd.timing('heap.total', usage.heapTotal);
+    statsd.timing('heap.used', usage.heapUsed);
+
+    if (usage.heapUsed > this.limit) {
+        console.log(usage.heapUsed, this.limit);
```
Comment (d00rman) [inline lib/heapwatch.js]: Why not use `this.logger.log()` (or remove this entirely since something is going to be logged either way) ?

**Comment by d00rman** [comment]
Awesome work @gwicke ! An extremely useful feature. I'd throw some more logic in it to dynamically adapt to the underlying running system. I'll create a ticket for that.

File: lib/heapwatch.js (hunk 1 above)
Comment (gwicke) [inline lib/heapwatch.js]: Good catch! This sneaked through from testing. Removed in #17.

**Pull request merged**